### PR TITLE
Enable depguard linter against github.com/pkg/errors

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

@simonpasquier @slashpai, I was thinking about adding https://github.com/fatih/faillint to our CI to make sure we're not using certain functions/packages. 

This could help us to fix https://github.com/prometheus-operator/prometheus-operator/issues/4682, and to make sure we don't re-introduce unwanted packages in the future

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by ArthurSens** [comment]
Now I just need to fix all those occurrences 😭

**Comment by simonpasquier** [comment]
Can't we use [depguard](https://golangci-lint.run/usage/linters/#depguard) which is available from golangci-lint?

**Comment by ArthurSens** [comment]
> Can't we use [depguard](https://golangci-lint.run/usage/linters/#depguard) which is available from golangci-lint?

Oh you even mentioned that in the issue and I missed 😬...

I guess depguard will catch a lot more problems, since I'm using faillint to catch only Wrap/Wrapf 😬

**Comment by ArthurSens** [comment]
I wonder if there is a way to make this one easier to review. 

Can golangci-lint only warn, but not fail, for a particular linter?

I feel like changing 700+ lines in a single PR is super easy to accidentally break some logs

**Comment by simonpasquier** [comment]
I agree that we need an incremental approach.

> Can golangci-lint only warn, but not fail, for a particular linter?

It looks like we can add  the enforcement  gradually by running it only for specific paths:
https://golangci-lint.run/usage/linters/#depguard

File: pkg/alertmanager/validation/v1alpha1/validation.go [hunk 1]
```diff
@@ -208,7 +207,7 @@ func validateEmailConfig(configs []monitoringv1alpha1.EmailConfig) error {
 		if config.Smarthost != "" {
 			_, _, err := net.SplitHostPort(config.Smarthost)
 			if err != nil {
-				return errors.Wrapf(err, "invalid field 'smarthost': %s", config.Smarthost)
+				return fmt.Errorf("invalid 'smarthost' %s: %w", config.Smarthost, err)
```
Comment (ArthurSens) [inline pkg/alertmanager/validation/v1alpha1/validation.go:210]: This is the only logline that I made a significant change. It already had `:`, so I re-arranged it.

(same in v1beta1)

**Comment by ArthurSens** [comment]
Okeee, started with 2 packages only. Will continue in other PRs.

File: pkg/alertmanager/validation/v1alpha1/validation.go (hunk 1 above)
Comment (simonpasquier) [reply pkg/alertmanager/validation/v1alpha1/validation.go:210]: this looks good to me.

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**